detector/vuls2/vuls2.go

```diff
@@ -0,0 +1,650 @@
+	advisoryTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/advisory"
+	criteriaTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/detection/condition/criteria"
+	criterionTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/detection/condition/criteria/criterion"
+	segmentTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/detection/segment"
+	ecosystemTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/detection/segment/ecosystem"
+	severityTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/severity"
+	v2 "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/severity/cvss/v2"
+	v31 "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/severity/cvss/v31"
+	vulnerabilityTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/vulnerability"
+	vcTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/vulnerability/content"
+	datasourceTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/datasource"
+	sourceTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/source"
+	db "github.com/MaineK00n/vuls2/pkg/db/common"
+	dbTypes "github.com/MaineK00n/vuls2/pkg/db/common/types"
+	"github.com/MaineK00n/vuls2/pkg/detect/ospkg"
+	detectTypes "github.com/MaineK00n/vuls2/pkg/detect/types"
+	scanTypes "github.com/MaineK00n/vuls2/pkg/scan/types"
+	"github.com/MaineK00n/vuls2/pkg/version"
+
+	"github.com/future-architect/vuls/config"
+	"github.com/future-architect/vuls/constant"
+	"github.com/future-architect/vuls/logging"
+	"github.com/future-architect/vuls/models"
+)
+
+var cveRe = regexp.MustCompile("CVE-[0-9]{4}-[0-9]+")
+
+// Detect detects vulnerabilities and fills ScanResult
+func Detect(r *models.ScanResult, vuls2Conf config.Vuls2Conf, noProgress bool) error {
+	switch r.Family {
+	case constant.RedHat, constant.CentOS, constant.Alma, constant.Rocky:
+	default:
+		return nil
+	}
+
+	if vuls2Conf.Repository == "" {
+		vuls2Conf.Repository = DefaultGHCRRepository
+	}
+	if vuls2Conf.Path == "" {
+		vuls2Conf.Path = DefaultPath
+	}
+
+	dbc, err := newDBConnection(vuls2Conf, noProgress)
+	if err != nil {
+		return xerrors.Errorf("Failed to get new db connection. err: %w", err)
+	}
+	if err := dbc.Open(); err != nil {
+		return xerrors.Errorf("Failed to open db. err: %w", err)
+	}
+	defer dbc.Close()
+
+	vuls2Scanned := preConvert(r)
+
+	vuls2Detected, err := detect(dbc, vuls2Scanned)
+	if err != nil {
+		return xerrors.Errorf("Failed to detect. err: %w", err)
+	}
+
+	e, err := ecosystemTypes.GetEcosystem(r.Family, r.Release)
+	if err != nil {
+		return xerrors.Errorf("Failed to get ecosystem. family: %s, release: %s, err: %w", r.Family, r.Release, err)
+	}
+
+	vulnInfos, err := postConvert(e, r.Family, vuls2Detected, vuls2Scanned)
+	if err != nil {
+		return xerrors.Errorf("Failed to post convert. err: %w", err)
+	}
+
+	r.ScannedCves = vulnInfos
```

Comment:
Are you throwing away vulnerabilities detected by DetectLibsCves?
https://github.com/future-architect/vuls/blob/894b30d925029a2d671fad686dd87ea806e29cf9/detector/detector.go#L49-L55